### ISSUE TITLE
default in encryption_enabled_by_default_for_room_type is misleading

### ISSUE THREAD

**Issue by Craeckie** [matrix-org/synapse#9794]
### Description

The option `encryption_enabled_by_default_for_room_type` as introduced in #7639 is misleading.

A default is something that has a value as long as it's not overwritten. In this case it would mean, that encryption is enabled for the specified room type as long as it is not disabled for a specific room/client/...
The current implementation prevents creation of unencrypted DMs/rooms. This is clearly not a _default_, but _forced_ encryption.

### Expected Naming

`force_encryption_enabled_for_room_type` or similar.

### Version information

`v1.31.0` and current `develop` branch.

**Comment by anoadragon453** [comment]
Indeed, the description at least should be updated, if not the name as well.

A similar conclusion was reached at the end of https://github.com/matrix-org/synapse/issues/9246, but nobody ended up creating an issue off the back of it. Thanks for doing so!

**Comment by FlorianJSa** [comment]
> force_encryption_enabled_for_room_type

Default is off so if you change this value it will overwrite the value

**Comment by anoadragon453** [comment]
Heads up that #9794 is an on-going, initial attempt at addressing this issue.

**Comment by richvdh** [comment]
itym #9983
